parkings/api/enforcement/valid_parking.py

```diff
@@ -29,9 +30,19 @@ class Meta:
             'is_disc_parking',
         ]
 
+    def __init__(self, *args, **kwargs):
+        super(ValidParkingSerializer, self).__init__(*args, **kwargs)
+        domain = self.context['request'].user.enforcer.enforced_domain
+        zone_field = self.fields['zone']
+        zone_qs = zone_field.get_queryset()
+        if zone_qs:
+            zone_qs.queryset = zone_qs.filter(domain=domain)
```

Comment:
Shouldn't this be `zone_field.queryset = ...`?

I'm pretty sure that it has a bug. Maybe add a test to prove it:
1. Add two zones: one to Domain D1 and another to Domain D2, both with the same code "Z"
2. Create a parking P1 to D1 with zone code Z
3. Check the zone FK of P1
4. Create a parking P2 to D2 with zone code Z
5. Check the zone FK of P2

Reply:
Oh, it was actually the valid_parking endpoint, not the parking creating endpoint (which I was talking about here). Well, then the test case should be almost the same, but instead of checking the created object, you should be checking that the valid_parking returns correct results when two parkings exists with the same zone code, but in different domains.

Reply:
Yes, it should be `zone_field.queryset` and not `zone_qs.queryset`.. Will fix this and add the tests.

Reply:
Bug fixed and tests added for both endpoints.